### PR TITLE
Issue 1577.2

### PR DESCRIPTION
**Pull request by rabbah**
Adds rules API handler test for use of `_` in action and trigger name. Reject missing namespace from either.

The second commit is a proposed fix for #1577 that permits unqualified action/trigger names in rule API operations which I think we should reject. The last commit adds additional tests to assert that a namespace must be specified (`_` is permitted as with the other referenced entities in sequences and bindings).

### CODE REVIEW

**Comment by rabbah** [comment]
pg2/664 approved.

**Comment by rabbah** [comment]
Dropped second commit: API requires fully qualified names (note `_` is allowed for namespace in the path).

File: tests/src/whisk/core/entity/test/SchemaTests.scala [hunk 1]
```diff
@@ -145,15 +133,24 @@ class SchemaTests extends FlatSpec with BeforeAndAfter with Matchers {
             JsObject("path" -> "a".toJson, "name" -> "b".toJson),
             JsObject("path" -> "a".toJson, "name" -> "b".toJson, "version" -> "0.0.1".toJson),
             JsString("a/b"),
-            JsObject("namespace" -> "a".toJson, "name" -> "b".toJson))
+            JsString("n/a/b"),
+            JsString("/a/b"),
+            JsString("/n/a/b"),
+            JsString("b")) //JsObject("namespace" -> "a".toJson, "name" -> "b".toJson))
```
Comment (cbickel) [inline tests/src/whisk/core/entity/test/SchemaTests.scala]: Is this entry used anywhere?

**Comment by rabbah** [comment]
Not now it isn't.

**Comment by markusthoemmes** [comment]
rebased, squashed, rephrased, merged, pushed and smiled in c95d7541f60d88716ef9dd2d5d975ac6df279110

**Pull request closed without merging**